python/ray/_private/gcs_aio_client.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from typing import Dict, List, Optional
+from concurrent.futures import ThreadPoolExecutor
+from ray._raylet import GcsClient
+from ray.core.generated import (
+    gcs_service_pb2,
+)
+import ray._private.utils
+
+
+logger = logging.getLogger(__name__)
+
+
+class AsyncProxy:
+    def __init__(self, inner, loop, executor):
+        self.inner = inner
+        self.loop = loop
+        self.executor = executor
+
+    def _function_to_async(self, func):
+        async def wrapper(*args, **kwargs):
+            return await self.loop.run_in_executor(self.executor, func, *args, **kwargs)
```

Comment:
is this thread safe?

Reply:
It should be thread safe as long as the wrapped func (i.e. the C++ PythonGcsClient class methods) is thread safe. cc @pcmoritz